pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -166,6 +202,36 @@ func (r *ReconcileWindowsMachineConfig) getCloudProvider(instance *wmcapi.Window
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling", "namespace", request.Namespace, "name", request.Name)
+	wmcoObject := false
+	// Fetch the Machine instance
+	Machine := &mapiv1.Machine{}
+	provisionedPhase := "Provisioned"
+	if err := r.client.Get(context.TODO(), request.NamespacedName, Machine); err != nil {
+		wmcoObject = true
+		// TODO: Uncomment this when we switch to Machine-api completely
+		//if k8sapierrors.IsNotFound(err) {
+		//	// Request object not found, could have been deleted after reconcile request.
+		//	// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+		//	// Return and don't requeue
+		//
+		//	return reconcile.Result{}, nil
+		//
+		//}
+		//// Error reading the object - requeue the request.
+		//return reconcile.Result{}, err
+	}
+	if !wmcoObject {
+		if Machine.Status.Phase == nil {
+			// Phase can be nil and should be ignored by WMCO
+			return reconcile.Result{}, nil
+		}
+		if *Machine.Status.Phase != provisionedPhase {
+			// If the Machine is not in provisioned state, WMCO shouldn't care about it
+			return reconcile.Result{}, nil
+		}
```

Comment:
Can't this be checked in the predicate?

Reply:
unfortunately no. i tried `&Machine{Status: Provisioned}` (don't remember the exact syntax) in predicate. Was having no effect.

Reply:
That just means the syntax was not accurate. It should definitely be possible given other [repos](https://github.com/operator-framework/operator-marketplace/blob/3d79e7f651d96bc57a4e5a335146ff5a9d363cbc/pkg/controller/operatorhub/operatorhub_controller.go#L52) are filtering based on things like `Name`.

Reply:
By the time we are in predicate we don't have the machine object associated, it will be generic runtime object, which we have to typeConvert.

Reply:
And is that hard to do?

Reply:
> unfortunately no. i tried `&Machine{Status: Provisioned}` (don't remember the exact syntax) in predicate. Was having no effect.

this was wrt https://github.com/openshift/windows-machine-config-operator/pull/78/files#diff-c9e016282d2886f8f7fc2355cc2c1bf4R129